stub/src/test/java/io/grpc/stub/ClientCallsTest.java

```diff
@@ -329,17 +345,16 @@ public void onCompleted() {
     integerStreamObserver.request(3);
     integerStreamObserver.onCompleted();
     assertTrue(latch.await(5, TimeUnit.SECONDS));
-    // Very that number of messages produced in each onReady handler call matches the number
+    // Verify that number of messages produced in each onReady handler call matches the number
     // requested by the client. Note that ClientCalls.asyncBidiStreamingCall will request(1)
     assertEquals(Arrays.asList(0, 1, 1, 2, 2, 2), receivedMessages);
   }
 
-  @org.junit.Ignore
   @Test
   public void inprocessTransportOutboundFlowControl() throws Exception {
     final CountDownLatch latch = new CountDownLatch(1);
-    final Semaphore semaphore = new Semaphore(1);
```

Comment:
It looks like it was to synchronize the callback with new call creation (with the acquire I removed below). But instead it just adds flakiness where the new thread created calls acquire first and then could have a total of three messages requested outstanding (if I hadn't also swapped to directExecutor). So I don't think it served an useful purpose, and was actually counter-productive because it introduced a race that wouldn't have been there otherwise.